### PR TITLE
ci: Exclude main branch when creating check

### PR DESCRIPTION
**Pull request by mayank99**
This should fix the failing build in main branch.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
           git-config-email: github-actions[bot]@users.noreply.github.com
 
       - name: Post gh pages link under Checks
-        if: always()
+        if: github.ref != 'refs/heads/main'
```
Comment (veekeys) [inline .github/workflows/build.yml:52]: but we need deploy results, if visual tests fail in main branch

Comment (bentleyvk) [inline .github/workflows/build.yml:52]: But it won't post if build fails on main? Or it really needs PR number or something similar for that to work?

**Review by bentleyvk — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (mayank99) [reply .github/workflows/build.yml:52]: we do
image

Comment (mayank99) [reply .github/workflows/build.yml:52]: it will
image

**Pull request merged**